.ci/pitest-survival-check-xml.groovy

```diff
@@ -107,7 +98,26 @@ private static int checkPitestReport(String profile, String flag) {
         final Node suppressedMutationNode = xmlParser.parse(suppressedMutationFileUri)
         suppressedMutations = getSuppressedMutations(suppressedMutationNode)
     }
-    return compareMutations(survivingMutations, suppressedMutations, flag)
+
+    final StringBuilder suppressionFileContent = new StringBuilder(1024)
+    suppressionFileContent.append('<?xml version="1.0" encoding="UTF-8"?>\n<suppressedMutations>')
+
+    survivingMutations.each {
+        suppressionFileContent.append(it.toXmlString())
+    }
+    if (survivingMutations.isEmpty()) {
+        suppressionFileContent.append('\n')
+    }
+    suppressionFileContent.append('</suppressedMutations>\n')
+
+    if (suppressionFile.exists()) {
+        suppressionFile.write(suppressionFileContent.toString())
+    } else if (!survivingMutations.isEmpty()) {
+        suppressionFile.createNewFile()
+        suppressionFile.write(suppressionFileContent.toString())
+    }
+
+    return compareMutations(survivingMutations, suppressedMutations)
```

Comment:
I am slightly confused. Why are we doing a compare AFTER we have written out the file? It seems to me this should be done first or be the thing that writes out the file.

Also, how come `suppressedMutations` is not taken into account when creating the suppression file?

Reply:
> I am slightly confused. Why are we doing a compare AFTER we have written out the file? It seems to me this should be done first or be the thing that writes out the file.

Doesn't matter, comparison shows a diff between what you currently have and what pit says. Suppressed mutations file is read before we make changes to it, so it doesn't make a difference.

> Also, how come suppressedMutations is not taken into account when creating the suppression file?

The suppression file is created from new surviving mutations, it does not matter what earlier was there in the file, the new report is reflected in the suppression file.

Reply:
I keep coming back to how we have this method called `compare` after we have already written the file.

I think it makes more sense to rename it to something like `printComparisonToConsole`.

Reply:
Done.